rules/naming/src/PropertyRenamer/AbstractPropertyRenamer.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Naming\PropertyRenamer;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\StaticPropertyFetch;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Stmt\Property;
+use PhpParser\Node\VarLikeIdentifier;
+use Rector\Core\PhpParser\NodeTraverser\CallableNodeTraverser;
+use Rector\Naming\Guard\DateTimeAtNamingConventionGuard;
+use Rector\Naming\Guard\GuardInterface;
+use Rector\Naming\Guard\HasMagicGetSetGuard;
+use Rector\Naming\Guard\NotPrivatePropertyGuard;
+use Rector\Naming\Guard\RamseyUuidInterfaceGuard;
+use Rector\Naming\RenameGuard\PropertyRenameGuard;
+use Rector\Naming\RenameGuard\RenameGuardInterface;
+use Rector\Naming\ValueObject\PropertyRename;
+use Rector\NodeNameResolver\NodeNameResolver;
+
+abstract class AbstractPropertyRenamer
+{
+    /**
+     * @var RenameGuardInterface
+     */
+    protected $propertyRenameGuard;
+
+    /**
+     * @var GuardInterface
+     */
+    protected $conflictingPropertyNameGuard;
```

Comment:
These `protected` properties usually means hidden dependency on sub-class, but we can handle that later

Reply:
These are used in all child classes that is why they are in the parent class. Do you mean even though they are in all child classes they should be moved to each of them?

Reply:
That's an assumption. There could be a child class with nullable property.

This would actually break in PHP 7.4 typed properties.

The correct way is to use contract via interface method and change protected property/method/constant to a contract guaranteed value